src/Servers/HttpSys/src/MessagePump.cs

```diff
@@ -214,11 +214,18 @@ private async Task ProcessRequestsWorker()
                 }
                 try
                 {
-                    ThreadPool.UnsafeQueueUserWorkItem(requestContext, preferLocal: false);
+                    if (_options.UnsafePreferInlineScheduling)
+                    {
+                        await requestContext.ExecuteAsync();
```

Comment:
@davidfowl thoughts on if I should await here or just ignore the result the the cases that request processing goes async?

Reply:
This is correct. This approach is inherently limited by the number of concurrent accept loops. 

cc @Tratcher